### DISCUSSION TITLE
Custom Fields values in ObjectVar

### DISCUSSION THREAD

**Discussion by allenderj** [netbox-community/netbox#8967] [Q&A]
Is it possible to use the values associated with a selection type custom field as the values in an ObjectVar in a custom script to be able to filter down other object vars in the same script?

**Comment by henchozd** [comment]
Hi,

Did you found a solution to your problem. I'm also trying to filter the output of my custom script using a value selected in the input form.

My custom field content : 

```
{
  "count": 1,
  "next": null,
  "previous": null,
  "results": [
    {
      "id": 1,
      "url": "https://127.0.0.1/api/extras/custom-fields/1/",
      "display": "Ordre de priorité pour gracefull shutdown",
      "content_types": [
        "dcim.device"
      ],
      "type": {
        "value": "select",
        "label": "Selection"
      },
      "object_type": null,
      "data_type": "string",
      "name": "ShutdownPriority",
      "label": "Ordre de priorité pour gracefull shutdown",
      "group_name": "",
      "description": "",
      "required": true,
      "search_weight": 1000,
      "filter_logic": {
        "value": "loose",
        "label": "Loose"
      },
      "ui_visibility": {
        "value": "read-write",
        "label": "Read/Write"
      },
      "default": "P0",
      "weight": 100,
      "validation_minimum": null,
      "validation_maximum": null,
      "validation_regex": "",
      "choices": [
        "P0",
        "P1",
        "P2",
        "P3",
        "P4",
        "P5",
        "P6",
        "P7",
        "P8"
      ],
      "created": "2022-12-19T10:22:45.021699Z",
      "last_updated": "2023-01-05T16:01:51.039017Z"
    }
  ]
}
```

Object var in my customscript : 

```
    shutdown_priority = ObjectVar(
        model=CustomField,
        required=False,
        description='Priority value e.g. P0'
    )
```

I didn't found the syntax to define the ObjectVar shutdown_priority to get the list of choices in the form 
I only get the list of the custom field in my input form, but I can't get the list of values of the custom field ...


Any help would be appreciated

**Comment by heidl89** [comment]
@henchozd Do you solved this already?

**Comment by Zombie-Toad** [comment]
I don't know if it helps but I figured out a solution to use custom field values in custom scripts.
https://github.com/netbox-community/netbox/discussions/18088
